py/ai/h2o/sparkling/ml/algos/H2ODeepLearning.py

```diff
@@ -16,53 +16,18 @@
 #
 
 from pyspark import keyword_only
-from pyspark.ml.util import JavaMLWritable, JavaMLReadable
-from pyspark.ml.wrapper import JavaEstimator
 
 from ai.h2o.sparkling import Initializer
-from ai.h2o.sparkling.ml.models import H2OMOJOModel
+from ai.h2o.sparkling.ml.algos.H2OAlgoBase import H2OAlgoBase
 from ai.h2o.sparkling.ml.params import H2ODeepLearningParams
-from ai.h2o.sparkling.ml.utils import set_double_values, validateEnumValue
-from ai.h2o.sparkling.sparkSpecifics import get_input_kwargs
 
 
-class H2ODeepLearning(H2ODeepLearningParams, JavaEstimator, JavaMLReadable, JavaMLWritable):
+class H2ODeepLearning(H2ODeepLearningParams, H2OAlgoBase):
 
     @keyword_only
-    def __init__(self, modelId=None, splitRatio=1.0, labelCol="label", weightCol=None, featuresCols=[], allStringColumnsToCategorical=True, columnsToCategorical=[],
-                 nfolds=0, keepCrossValidationPredictions=False, keepCrossValidationFoldAssignment=False, parallelizeCrossValidation=True,
-                 seed=-1, distribution="AUTO", epochs=10.0, l1=0.0, l2=0.0, hidden=[200,200], reproducible=False,
-                 convertUnknownCategoricalLevelsToNa=False, foldCol=None, predictionCol="prediction", detailedPredictionCol="detailed_prediction",
-                 withDetailedPredictionCol=False, convertInvalidNumbersToNa=False, **deprecatedArgs):
+    def __init__(self, **kwargs):
```

Comment:
I really like the compactness of the code, but wouldn't users loss an opportunity to see IDE hints with the list of possible constructor arguments?

Reply:
Yup, will keep the params in init method